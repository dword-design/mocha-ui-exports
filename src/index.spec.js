import { endent, mapValues } from '@dword-design/functions'
import execa from 'execa'
import outputFiles from 'output-files'
import withLocalTmpDir from 'with-local-tmp-dir'

const runTest = config => () =>
  withLocalTmpDir(async () => {
    config = { args: [], ...config }
    await outputFiles(config.files)
    const output = await execa(
      'mocha',
      ['--ui', require.resolve('.'), ...config.args, '*.spec.js'],
      { all: true }
    )
    expect(output.all).toMatch(config.regex)
  })

export default {
  'after test': {
    files: {
      'index.spec.js': endent`
        module.exports = {
          after: () => console.log('foo'),
          bar: () => {},
        }
      `,
    },
    regex: new RegExp(endent`
      ^

        index
          . bar
      foo


        1 passing \\(.*?\\)
      $
    `),
  },
  'beforeEach test': {
    files: {
      'index.spec.js': endent`
        module.exports = {
          beforeEach: () => console.log('foo'),
          bar: () => {},
          baz: () => {},
        }
      `,
    },
    regex: new RegExp(endent`
      ^

        index
      foo
          . bar
      foo
          . baz


        2 passing \\(.*?\\)
      $
    `),
  },
  describe: {
    files: {
      'index.spec.js': endent`
        module.exports = {
          foo: {
            bar: () => {},
          },
        }
      `,
    },
    regex: new RegExp(endent`
      ^

        index
          foo
            . bar


        1 passing \\(.*?\\)
      $
    `),
  },
  'multiple suites': {
    files: {
      'cli.spec.js': endent`
        module.exports = {
          bar: () => {},
        }
      `,
      'index.spec.js': endent`
        module.exports = {
          foo: () => {},
        }
      `,
    },
    regex: new RegExp(endent`
      ^

        cli
          . bar

        index
          . foo


        2 passing \\(.*?\\)
      $
    `),
  },
  'multiple tests': {
    files: {
      'index.spec.js': endent`
        module.exports = {
          afterEach: () => console.log('foo'),
          bar: () => {},
          baz: () => {},
        }
      `,
    },
    regex: new RegExp(endent`
      ^

        index
          . bar
      foo
          . baz
      foo


        2 passing \\(.*?\\)
      $
    `),
  },
  'non-test file': {
    args: ['--file', 'before.js'],
    files: {
      'before.js': endent`
        module.exports = {
          before: () => console.log('foo'),
        }
      `,
      'index.spec.js': endent`
        module.exports = {
          bar: () => {},
          baz: () => {},
        }
      `,
    },
    regex: new RegExp(endent`
      ^

      foo
        index
          . bar
          . baz


        2 passing \\(.*?\\)
      $
    `),
  },
  valid: {
    files: {
      'index.spec.js': endent`
        module.exports = {
          foo: () => {},
        }
      `,
    },
    regex: new RegExp(endent`
      ^

        index
          . foo


        1 passing \\(.*?\\)
      $
    `),
  },
} |> mapValues(runTest)
