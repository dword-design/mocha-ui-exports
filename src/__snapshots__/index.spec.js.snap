// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index after test 1`] = `
"

  index
    ✓ bar
foo


  1 passing
"
`;

exports[`index array 1`] = `
"

  index
    ✓ 0


  1 passing
"
`;

exports[`index beforeEach test 1`] = `
"

  index
foo
    ✓ bar
foo
    ✓ baz


  2 passing
"
`;

exports[`index describe 1`] = `
"

  index
    foo
      ✓ bar


  1 passing
"
`;

exports[`index multiple suites 1`] = `
"

  cli
    ✓ bar

  index
    ✓ foo


  2 passing
"
`;

exports[`index multiple tests 1`] = `
"

  index
    ✓ bar
foo
    ✓ baz
foo


  2 passing
"
`;

exports[`index non-test file 1`] = `
"

foo
  index
    ✓ bar
    ✓ baz


  2 passing
"
`;

exports[`index valid 1`] = `
"

  index
    ✓ foo


  1 passing
"
`;
